nasa/ow_simulator
# Oceanwater 1070 - Accurate depth on secondary trench passes

## Linked Issues:
| EPIC ⚡ | [OCEANWATER-1069](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1069) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-1070](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1070) |

This PR fixes secondary trenching passes resulting in the wrong depth (too deep) being passed to the Balovnev model. This would happen because some amount of the edge of the trench is always modified in the secondary pass. By ignoring the edge pixels in the differential image, this has been fixed.

## Summary of Changes
* ~20% margins are cut away from the modification differential image before the maximum pixel value is computed.~
* 30% of scoop-aligned margins are ignored when the maximum pixel magnitude is computed from the differential image.
* Some minor code improvements  

## Test
1. Launch atacama_y1a
2. Call grind in the default area. 
`rosrun ow_lander task_grind.py`
3. Bring up a plot of depth in RQT. You can use the **balovnev_analysis.perspective** in _ow/config_ for this. Axis boundaries may have to be adjusted. 
4. Call a dig that barely scratches the surface. 
`rosrun ow_lander task_scoop_linear.py -z 0 -d 0.1`
5. Dig exactly 4 cm more. Check the depth plot. A new plateau will appear. Its maximum value should be ~0.04.
`rosrun ow_lander task_scoop_linear.py -z 0 -d 0.14`
6. Dig exactly 2 cm more. The new plateau's maximum value should be ~0.02.
`rosrun ow_lander task_scoop_linear.py -z 0 -d 0.16`

The following image are my results of the previous 2 digs. Note one is about 4cm in height and the other is about 2cm in height. Your plot may not look exactly the same, but the heights should be similar. 
![image](https://user-images.githubusercontent.com/17039973/220228061-4bf344a8-25e9-4afa-8493-c62e3481a813.png)
